### PR TITLE
update example of AST for TokenTypes.PARAMETERS

### PR DESCRIPTION
**Pull request by Venkatesh-24**
Closes: #9551

Screenshot from 2021-11-01 23-43-51

```
**Test.java**
public class Test { 
  void foo(int firstParameter, int... secondParameter) {}
}

**AST**
java -jar checkstyle-9.0.1-all.jar -t Test.java
`--CLASS_DEF -> CLASS_DEF [4:0]
    |--MODIFIERS -> MODIFIERS [4:0]
    |   `--LITERAL_PUBLIC -> public [4:0]
    |--LITERAL_CLASS -> class [4:7]
    |--IDENT -> Test [4:13]
    `--OBJBLOCK -> OBJBLOCK [4:18]
        |--LCURLY -> { [4:18]
        |--METHOD_DEF -> METHOD_DEF [5:2]
        |   |--MODIFIERS -> MODIFIERS [5:2]
        |   |--TYPE -> TYPE [5:2]
        |   |   `--LITERAL_VOID -> void [5:2]
        |   |--IDENT -> foo [5:7]
        |   |--LPAREN -> ( [5:10]
        |   |--PARAMETERS -> PARAMETERS [5:11]
        |   |   |--PARAMETER_DEF -> PARAMETER_DEF [5:11]
        |   |   |   |--MODIFIERS -> MODIFIERS [5:11]
        |   |   |   |--TYPE -> TYPE [5:11]
        |   |   |   |   `--LITERAL_INT -> int [5:11]
        |   |   |   `--IDENT -> firstParameter [5:15]
        |   |   |--COMMA -> , [5:29]
        |   |   `--PARAMETER_DEF -> PARAMETER_DEF [5:31]
        |   |       |--MODIFIERS -> MODIFIERS [5:31]
        |   |       |--TYPE -> TYPE [5:31]
        |   |       |   `--LITERAL_INT -> int [5:31]
        |   |       |--ELLIPSIS -> ... [5:34]
        |   |       `--IDENT -> secondParameter [5:38]
        |   |--RPAREN -> ) [5:53]
        |   `--SLIST -> { [5:55]
        |       `--RCURLY -> } [5:56]
        `--RCURLY -> } [6:0]

# Printing the code that we care
java -jar checkstyle-9.0.1-all.jar -t Test.java | grep 5:
        |--METHOD_DEF -> METHOD_DEF [5:2]
        |   |--MODIFIERS -> MODIFIERS [5:2]
        |   |--TYPE -> TYPE [5:2]
        |   |   `--LITERAL_VOID -> void [5:2]
        |   |--IDENT -> foo [5:7]
        |   |--LPAREN -> ( [5:10]
        |   |--PARAMETERS -> PARAMETERS [5:11]
        |   |   |--PARAMETER_DEF -> PARAMETER_DEF [5:11]
        |   |   |   |--MODIFIERS -> MODIFIERS [5:11]
        |   |   |   |--TYPE -> TYPE [5:11]
        |   |   |   |   `--LITERAL_INT -> int [5:11]
        |   |   |   `--IDENT -> firstParameter [5:15]
        |   |   |--COMMA -> , [5:29]
        |   |   `--PARAMETER_DEF -> PARAMETER_DEF [5:31]
        |   |       |--MODIFIERS -> MODIFIERS [5:31]
        |   |       |--TYPE -> TYPE [5:31]
        |   |       |   `--LITERAL_INT -> int [5:31]
        |   |       |--ELLIPSIS -> ... [5:34]
        |   |       `--IDENT -> secondParameter [5:38]
        |   |--RPAREN -> ) [5:53]
        |   `--SLIST -> { [5:55]
        |       `--RCURLY -> } [5:56]
```
**Expected update for javadoc**

``` 
     * METHOD_DEF -&gt; METHOD_DEF
     *  |--MODIFIERS -&gt; MODIFIERS
     *  |--TYPE -&gt; TYPE
     *  |   `--LITERAL_VOID -&gt; void
     *  |--IDENT -&gt; foo
     *  |--LPAREN -&gt; (
     *  |--PARAMETERS -&gt; PARAMETERS
     *  |   |--PARAMETER_DEF -&gt; PARAMETER_DEF
     *  |   |   |--MODIFIERS -&gt; MODIFIERS
     *  |   |   |--TYPE -&gt; TYPE
     *  |   |   |   `--LITERAL_INT -&gt; int
     *  |   |   `--IDENT -&gt; firstParameter
     *  |   |--COMMA -&gt; ,
     *  |   `--PARAMETER_DEF -&gt; PARAMETER_DEF
     *  |       |--MODIFIERS -&gt; MODIFIERS
     *  |       |--TYPE -&gt; TYPE
     *  |       |   `--LITERAL_INT -&gt; int
     *  |       |--ELLIPSIS -&gt; ...
     *  |       `--IDENT -&gt; secondParameter
     *  |--RPAREN -&gt; )
     *      `--SLIST -&gt; {
     *          `--RCURLY -&gt; }
```

### CODE REVIEW

**Comment by Venkatesh-24** [comment]
Please review it mentors.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -681,28 +681,59 @@
...
-     *     |       |--MODIFIERS -&gt; MODIFIERS
-     *     |       |--TYPE -&gt; TYPE
-     *     |       |   `--LITERAL_INT -&gt; int
-     *     |       |--ELLIPSIS -&gt; ...
-     *     |       `--IDENT -&gt; secondParameter
-     *     |--RPAREN -&gt; )
-     *      `--SLIST -&gt; {
-     *          `--RCURLY -&gt; }
+     *  |--METHOD_DEF -> METHOD_DEF
+        |   |--MODIFIERS -> MODIFIERS
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:685]: Remove this `|` and make leading `*`

**Review by romani — CHANGES_REQUESTED** [review]
Items:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (Venkatesh-24) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:685]: Yeah Sure.

Comment (Venkatesh-24) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: I have changed it. Please review it @romani

**Comment by romani** [comment]
Please review CI executions and resolve failures.
Make sure `mvn verify` passing on your local before you push

**Comment by Venkatesh-24** [comment]
@romani Please share some links that contains about what are all the commands that I need to execute before pushing the code in order to pass all the checks.

**Comment by Rahulkhinchi03** [comment]
@Venkatesh-24 Please run mvn clean verify.

**Comment by suman-somu** [comment]
@Venkatesh-24  here is the commands, apply according to your commit.

**Comment by suman-somu** [comment]
@Venkatesh-24, [here](https://checkstyle.sourceforge.io/contributing.html#Quality_matters) is the list . 
Apply according to your commit.

**Comment by Venkatesh-24** [comment]
I ran mvn clean verify i am getting cannot resolve to a type java language lexer. I dont know what to do now. Please someone guide me.

**Comment by Venkatesh-24** [comment]
@romani I ran maven clean verify command and the build was successful. Please review it

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -681,28 +681,59 @@
...
+     *      |--VARIABLE_DEF -> VARIABLE_DEF
+     *      |   |--MODIFIERS -> MODIFIERS
+     *      |   |--TYPE -> TYPE
+     *      |   |   `--LITERAL_INT -> int
+     *      |   |--IDENT -> sum
+     *      |   `--ASSIGN -> =
+     *      |       `--EXPR -> EXPR
+     *      |           `--IDENT -> num1
+     *      |--SEMI -> ;
+     *      |--LITERAL_FOR -> for
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:715]: Please use AST for example that you have in code tag

**Review by romani — CHANGES_REQUESTED** [review]
Item

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (Venkatesh-24) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:715]: @romani Do you mean that I should update AST along with line number in TokenTypes.java

**Review by Rahulkhinchi03 — COMMENTED** [review]
Remove this `|--`  from starting and lead with  `*`

**Comment by Venkatesh-24** [comment]
Please review it.@romani

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Put in file `void foo(int firstParameter, int... secondParameter) {}`  and copy AST only  for this line

**Comment by romani** [comment]
please make commit message like:
` Issue #9551: update example of AST for TokenTypes.PARAMETERS `

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 2 above)
Comment (Venkatesh-24) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: Okk I will do it

**Comment by Venkatesh-24** [comment]
@romani I have made the changes but wercker/build is failing. I dont know what to do.

**Review by romani — APPROVED** [review]

**Comment by Venkatesh-24** [comment]
Thank you @romani, @Rahulkhinchi03, @suman-somu,  for guiding me through my first pull request in CheckStyle

**Pull request merged**